### DISCUSSION TITLE
Does this action only work on Public Repo's?

### DISCUSSION THREAD

**Discussion by mchittineni** [github-community-projects/contributors#161] [Q&A] [answered]
Does this action only work on public repositories? When I tried using it with our organization, it only processed a public repository and generated a report for it. However, when I attempted to run the action on a specific private repository within the organization, it failed, returning the following error:

`Traceback (most recent call last):
  File "/action/workspace/contributors.py", line 182, in <module>
    main()
  File "/action/workspace/contributors.py", line 36, in main
    contributors = get_all_contributors(
                   ^^^^^^^^^^^^^^^^^^^^^
  File "/action/workspace/contributors.py", line [10](https://github.com/Uni-of-Exeter/digital.devops.sandpit/actions/runs/10390082883/job/28769708850#step:4:11)4, in get_all_contributors
    owner, repo_name = repo.split("/")
    ^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 1)`

Here is my complete workflow file:

`
    steps:

    - name: Get dates for last month
      run: |
        # Calculate the first date of the previous month
        first_date=$(date -d "last month" +%Y-%m-01)

        # Calculate the last date of the previous month
        last_date=$(date -d "$first_date +1 month -1 day" +%Y-%m-%d)

        # Set an environment variable with the date range
        echo "START_DATE=$first_date" >> "$GITHUB_ENV"
        echo "END_DATE=$last_date" >> "$GITHUB_ENV"
        echo "last_month=$first_date..$last_date" >> "$GITHUB_ENV"

    - name: Run contributor action
      uses: github/contributors@v1
      env:
        GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
        START_DATE: ${{ env.START_DATE }}
        END_DATE: ${{ env.END_DATE }}
        REPOSITORY: ${{ github.event.repository.name }}
  
    - name: Create Report for GitHub Contributors
      uses: peter-evans/create-issue-from-file@v5
      with:
        title: Monthly Contributor's Report
        token: ${{ secrets.GITHUB_TOKEN }}
        content-filepath: ./contributors.md
        assignees: xxxxx`

**Comment by zkoppert** [comment, marked as answer]
Can you check that github.event.repository.name has a value and its format is owner/repo? More specifically I suspect that the value has no / character in it. 

The stack trace leads me to believe that is where the problem is occurring for you.

**Reply by mchittineni** [reply]
Hi, it's returning a correct value (i.e name of the repo only without the org. name).

**Reply by zkoppert** [reply]
The action is expecting to see the org name and the slash in there as well. Can you choose a different variable or manually put the full name (owner/repo) in there. That is why the code is failing to execute. 

For example:

`REPOSITORY: github/contributors`

**Reply by mchittineni** [reply]
i just gave a try with the org/repo name

and this is. what it cam up with

`Error getting contributors for repository: xxxxx/yyyyyy
403 Resource not accessible by integration`

`Error getting contributors for repository: xxxxx/yyyyyy
403 Resource not accessible by integration`


![Screenshot 2024-08-15 at 16 21 06](https://github.com/user-attachments/assets/d596feab-cb6e-45a2-b83b-2467a8b6324e)

Even though there was a error, the issue was created with a empty metric as follows:

![Screenshot 2024-08-15 at 16 25 18](https://github.com/user-attachments/assets/efa14dfd-ccb6-4a51-a508-d44d689d575f)

**Reply by mchittineni** [reply]
got it, the workflow needed a PAT Token to make it work. When i have added the PAT to the workflow it gave me the metrics as expected.

![Screenshot 2024-08-15 at 16 52 20](https://github.com/user-attachments/assets/c173ddaa-b3d2-4d6c-b714-6a50c0ab0003)

**Comment by zkoppert** [comment]
Wohoo! Glad its working!

**Reply by mchittineni** [reply]
Even without a PAT Token, a GitHub Token with these permissions will also work
```
permissions:
  issues: write
  pull-requests: read
  contents: read
```
